keep/providers/checkmk_provider/checkmk_provider.py

```diff
@@ -0,0 +1,102 @@
+"""
+Checkmk is a monitoring tool for Infrastructure and Application Monitoring.
+"""
+
+from keep.api.models.alert import AlertDto, AlertStatus, AlertSeverity
+from keep.contextmanager.contextmanager import ContextManager
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig
+
+class CheckmkProvider(BaseProvider):
+  """Get alerts from Checkmk into Keep"""
+
+  webhook_description = ""
+  webhook_template = ""
+  webhook_markdown = """
+  💡 For more details on how to configure Checkmk to send alerts to Keep, see the [Keep documentation](https://docs.keephq.dev/providers/documentation/checkmk-provider).
+  1. Checkmk supports custom notification scripts.
+  2. Install Keep webhook script following the [Keep documentation](https://docs.keephq.dev/providers/documentation/checkmk-provider).
+  3. In Checkmk WebUI, go to Setup.
+  4. Click on Add rule.
+  5. In the Notifications method section, select Webhook - KeepHQ and choose "Call with the following parameters:".
+  6. Configure the Rule properties, Contact selections, and Conditions according to your requirements.
+  7. The first parameter is the Webhook URL of Keep which is {keep_webhook_api_url}.
+  8. The second parameter is the API Key of Keep which is {api_key}.
+  9. Click on Save.
+  10. Now Checkmk will be able to send alerts to Keep.
+  """
+
+  SEVERITIES_MAP = {
+    "OK": AlertSeverity.INFO,
+    "WARN": AlertSeverity.WARNING,
+    "CRIT": AlertSeverity.CRITICAL,
+    "UNKNOWN": AlertSeverity.INFO,
+  }
+
+  STATUS_MAP = {
+    "UP": AlertStatus.RESOLVED,
+    "DOWN": AlertStatus.FIRING,
+    "UNREACH": AlertStatus.FIRING
+  }
+
+  PROVIDER_DISPLAY_NAME = "Checkmk"
+  PROVIDER_TAGS = ["alert"]
+  
+  def __init__(
+      self, context_manager: ContextManager, provider_id: str, config: ProviderConfig
+  ):
+    super().__init__(context_manager, provider_id, config)
+
+  def validate_config():
+    """
+    No validation required for Checkmk provider.
+    """
+    pass
+
+  @staticmethod
+  def _format_alert(event: dict, provider_instance: BaseProvider = None) -> AlertDto | list[AlertDto]:
+    def _check_values(value):
+      if value not in event or event[value] == '':
+        return None
+      return event[value]
+    
+    # based on the status of the event, set the severity
+    def _set_severity(status):
+      if status == "UP":
+        return AlertSeverity.INFO
+      elif status == "DOWN":
+        return AlertSeverity.CRITICAL
+      elif status == "UNREACH":
+        return AlertSeverity.CRITICAL
```

Comment:
what is it for? what's the difference between this and SEVERITIES_MAP? add a comment on it